mlflow/tracking/client.py

```diff
@@ -2116,7 +2116,12 @@ def create_model_version(
             Stage: None
         """
         tracking_uri = self._tracking_client.tracking_uri
-        if not run_link and is_databricks_uri(tracking_uri) and tracking_uri != self._registry_uri:
+        if (
+            not run_link
+            and run_id
```

Comment:
i don't think this should be here - do we need it for some test to pass?

Reply:
Ah I don't think it's necessary. I added it because `run_link` wouldn't be set without a non-null `run_id` - will remove this filter

Reply:
ahhh i see... hm.. actually we may want to throw something useful in this case - let me take a look briefly.